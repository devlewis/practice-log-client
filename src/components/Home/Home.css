.container_home {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.rgstrbutton {
  margin-top: 50px;
}
