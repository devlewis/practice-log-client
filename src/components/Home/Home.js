import React, { Component } from "react";
import "../Home/Home.css";
import LoginForm from "../LoginForm/LoginForm";
import { withRouter } from "react-router";
import Context from "../../Context";

class Home extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  };

  static contextType = Context;

  onClickRegister = () => {
    this.props.history.push("/register");
  };

  handleLoginSuccess = () => {
    this.context.onHandleLoginFetch(this.props.history);
  };

  render() {
    return (
      <div className="container_home">
        <h2>PRACTICE LOG HOME PAGE</h2>
        <div className="container_login">
          <LoginForm onLoginSuccess={this.handleLoginSuccess} />
        </div>
        <button className="rgstrbutton" onClick={this.onClickRegister}>
          Make a new account!
        </button>
      </div>
    );
  }
}

export default withRouter(Home);
