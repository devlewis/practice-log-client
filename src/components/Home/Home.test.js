import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Home from "./Home";

describe(`Home component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders form.Home by default", () => {
    const wrapper = shallow(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Home given props", () => {
    const wrapper = shallow(<Home {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
