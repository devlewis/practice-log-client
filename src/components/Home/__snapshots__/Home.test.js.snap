// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders form.Home by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Home component renders the Home given props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
